_rules/printable-characters-shortcut-ffbc54.md

```diff
@@ -0,0 +1,547 @@
+---
+id: ffbc54
+name: No keyboard shortcuts use only printable characters
+rule_type: atomic
+description: |
+  This rule checks that if keyboard shortcuts are implemented using only printable characters, then there is a mechanism to disable the shortcut, or to remap the shortcut to use one or more non-printable characters keys, or the shortcut for a user interface component is only available when that component has focus.
+accessibility_requirements:
+  wcag21:2.1.4: # Character Key Shortcuts (A)
+    forConformance: true
+    failed: not satisfied
+    passed: satisfied
+    inapplicable: further testing needed
+acknowledgements:
+  authors:
+    - João Vicente
+    - Carlos Duarte
+---
+
+## Applicability
+
+The rule applies to any [keyboard event][]: 
```

Comment:
Thinking about the composition of the rule and I'm now running into this problem…

If we target the (DOM) keyboard event, then the (**focus**) condition is OK, but the (**disable**)/(**remap**) isn't really…
1. I press 'a'. A keyboard event fires and does stuff (bad).
2. I activate some UI control.
3. I press 'a'. A keyboard event fires but is now disabled by the previous step.

Now, the problem is that steps 1. and 3. are **two different** keyboard events. So the first one is actually failing the rule because we can't go back in time and block it.

I think that the (**disable**)/(**remap**) condition can be rephrased as blocking future events with the same `key` attribute and dispatched to the same event target.

Going that direction does cause a risk of needing to check stuff in an unspecified distant future. Which is actually sort of already present:
If a button disable the 'a' shortcut for the next hour (and automatically re-enable it afterwards), is it passing the rule? the SC? should it be?
This is a bit different from the "blocking" duration (nothing happen <1min after firing the event) but somewhat similar: how long am I supposed to watch the page to decide that the event has been disabled?

Reply:
There's already an issue open to discuss the question of how long the event needs to remain disabled.

On the other topic, I don't agree that the first event fails the rule if in the future (i.e. the process of testing the rule) we discover that there is an instrument to disable or remap the shortcut. That's what the rule states.